pilot/pkg/model/context.go

```diff
@@ -277,10 +307,15 @@ func GetProxyConfigNamespace(proxy *Proxy) string {
 	}
 
 	// First look for ISTIO_META_CONFIG_NAMESPACE
-	if configNamespace, found := proxy.Metadata["CONFIG_NAMESPACE"]; found {
+	if configNamespace, found := proxy.Metadata[pilot.ConfigNamespace]; found {
 		return configNamespace
 	}
 
+	parts := strings.Split(proxy.ID, ".")
+	if len(parts) > 1 {
+		return parts[1]
+	}
+
```

Comment:
you dont need this.. The config namespace is available for all proxies as its added through the sidecar injector. Also, we have no way of knowing if parts[1] is a namespace or not (example.us.com) - as this proxy may not be running in kubernetes. It is for this reason that I explicitly choose to return no namespace so that the code continues to work for older proxies on non-k8s